source/dub/compilers/ldc.d

```diff
@@ -150,8 +150,15 @@ config    /etc/ldc2.conf (x86_64-pc-linux-gnu)
 			settings.lflags = null;
 		}
 
-		if (settings.options & BuildOption.pic)
-			settings.addDFlags("-relocation-model=pic");
+		if (settings.options & BuildOption.pic) {
+			if (platform.isWindows()) {
+				// This has nothing to do with PIC, but as it's implicitly/commonly used to
+				// compile a shared library, explicitly specify what `-shared` defaults to.
+				settings.addDFlags("-fvisibility=public", "-dllimport=all");
```

Comment:
If we go through with this change, it might be about time I did a DIP to replace export. Might as well make the change at the language level if this is what is actually required to make shared libraries work correctly.

@ibuclaw do any changes need to be made to match this with GDC?

Reply:
Replace it with what? `export` is IMO still very useful to explicitly export a symbol when compiling with (implicit or explicit) `-fvisibility=hidden`. And there's a rather new `@ldc.attributes.hidden` UDA to complement it on a by-symbol basis, which should probably be standardized in `core.attribute`, at least with GDC too.

Reply:
By default everything is exported if it's visible outside a module (so non-private). If you want fine-grained control you use a pragma (although a UDA works too).

This would match these changes and how people actually use it in practice.

Reply:
IDK, to me the status quo (with LDC) in this regard seems totally fine - with `-fvisibility`, `export` and (exotic) `@hidden`.